### PR TITLE
[Macro] Remove limitation for types of expressions accepted as named arguments

### LINKED ISSUE #5808: Subquery passed as named argument to a macro results in binder error
### What happens?

A subquery passed as named argument to a macro results in a binder error.

### To Reproduce

```
-- Create macro f that takes an argument x and adds one to it
-- If no argument is provided, default value NULL is used for argument x
CREATE MACRO f(x := NULL) AS x+1;

-- ✓
SELECT f();

-- ✓
SELECT f(x := 41);

-- ⚠ This statement fails with:
-- Binder Error: Macro function 'f()' requires 0 positional arguments, but a single positional argument was provided.
SELECT f(x := (SELECT 41));
```

### OS:

ArchLinux

### DuckDB Version:

v0.6.1 and v0.6.2-dev910 e74e2471a5

### DuckDB Client:

CLI

### Full Name:

Chris

### Affiliation:

N/A

### Have you tried this on the latest `master` branch?

- [X] I agree

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] I agree

### PR DESCRIPTION
**Pull request by Tishj**
This PR fixes #5808 

Seems like an easy fix, I do wonder why this limitation was there, as it did not break any tests.
@lnkuiper do you know why we had this limitation?
If so I'll add some tests for them and just add subquery to the accepted types instead.

Is the limitation somehow related to the "alias" part of it?
Are there types of expressions that use the alias for something, that we don't want to mistakenly see as a named parameter?

### CODE REVIEW

File: test/sql/catalog/function/test_macro_default_arg.test [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# name: test/sql/catalog/function/test_macro_default_arg.test
+# group: [function]
+
+#statement ok
+#pragma enable_verification;
```
Comment (lnkuiper) [inline test/sql/catalog/function/test_macro_default_arg.test:5]: Does this need to be uncommented again?

**Review by lnkuiper — CHANGES_REQUESTED** [review]
I don't remember exactly why I added this limitation. I think it's because you can make some really weird macros if you allow expressions there. Can you try to create a few more tests? For example:
```sql
-- default param references positional param
create macro my_macro1(a, b := a) as a + b;
-- default param references column of table within subquery expression
create macro my_macro2(a := i) as (select min(a) from integers);
```

These are just two tests that come to mind, but maybe there are more weird default params that you can try.

File: test/sql/catalog/function/test_macro_default_arg.test [hunk 2]
```diff
@@ -0,0 +1,71 @@
+# name: test/sql/catalog/function/test_macro_default_arg.test
+# group: [function]
+
+statement ok
+pragma enable_verification;
+
+#statement ok
+#CREATE MACRO f(x := NULL) AS x+1;
```
Comment (Mytherin) [inline test/sql/catalog/function/test_macro_default_arg.test:8]: Is this meant to be commented out?

**Comment by Tishj** [comment]
I'm not really sure what type of tests would be needed for this if I'm honest.
I get that removing this limitation is potentially error-prone, especially when we can't figure out why the limitation was there in the first place.
I could just add this type of expression to the whitelist instead of removing the entire limitations altogether, would that be preferable?

**Comment by Mytherin** [comment]
I think the current solution is fine, @lnkuiper thoughts?

**Comment by lnkuiper** [comment]
I agree. This can be merged

**Pull request merged**